src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/ServiceContext.cs

```diff
@@ -7,13 +7,13 @@
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal
 {
-    public class ServiceContext
+    public class ServiceContext<TRequestHandler> where TRequestHandler : struct, IHttpHeadersHandler, IHttpRequestLineHandler
```

Comment:
Why is this generic? Just make it have the func be `Func<FrameAdapter, IHttpParser<FrameAdapter>>`